app/src/main/java/com/example/composesigninsample/MainActivity.kt

```diff
@@ -49,11 +60,35 @@ fun AppRouter() {
     val navController = rememberNavController()
     NavHost(navController = navController, startDestination = Screen.HomeScreen.route) {
         composable(Screen.HomeScreen.route) {
-            HomeScreen({
-                navController.navigate(Screen.SignInScreen.route)
-            })
+            val homeViewModel: HomeViewModel = hiltViewModel()
+
+            HomeScreen(
+                navigateToSignIn = {
+                    navController.navigate(Screen.SignInScreen.route) {
+                        popUpTo(Screen.HomeScreen.route) {
+                            inclusive = true
+                        }
```

Comment:
Just so I'm understanding correctly. This means that when I'm on the HomeScreen, and I navigate to signIn, the homeScreen is immediately popped off, right? Makes sense to me, but pop* always confuses me.

Reply:
Yes, if you navigate from home to signin, then home is popped off